waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/model/AlluxioMetastoreFilter.java

```diff
@@ -0,0 +1,131 @@
+import org.apache.hadoop.hive.metastore.api.StorageDescriptor;
+import org.apache.hadoop.hive.metastore.api.Table;
+
+/**
+ * For testing purposes
+ * */
+public class AlluxioMetastoreFilter implements MetaStoreFilterHook {
+
+  public static final String ALLUXIO_PREFIX = "alluxio://";
+
+  public AlluxioMetastoreFilter(HiveConf conf) {
+  }
+
+  @Override
+  public List<String> filterDatabases(List<String> dbList) throws MetaException {
+    return dbList;
+  }
+
+  @Override
+  public Database filterDatabase(Database dataBase) throws MetaException, NoSuchObjectException {
+    return dataBase;
+  }
+
+  @Override
+  public List<String> filterTableNames(String dbName, List<String> tableList) throws MetaException {
+    return tableList;
+  }
+
+  @Override
+  public Table filterTable(Table table) throws MetaException, NoSuchObjectException {
+    setAlluxioLocation(table);
+    return table;
+  }
+
+  @Override
+  public List<Table> filterTables(List<Table> tableList) throws MetaException {
+    for (Table table: tableList){
+      setAlluxioLocation(table);
+    }
+    return tableList;
+  }
+
+  @Override
+  public List<Partition> filterPartitions(List<Partition> partitionList) throws MetaException {
+    for (Partition partition: partitionList){
+      setAlluxioLocation(partition.getSd());
+    }
+    return partitionList;
+  }
+
+  @Override
+  public List<PartitionSpec> filterPartitionSpecs(List<PartitionSpec> partitionSpecList) throws MetaException {
+    for (PartitionSpec partitionSpec : partitionSpecList) {
+      setAlluxioLocation(partitionSpec.getSharedSDPartitionSpec().getSd());
+      filterPartitions(partitionSpec.getPartitionList().getPartitions());
+    }
+    return partitionSpecList;
+  }
+
+  @Override
+  public Partition filterPartition(Partition partition) throws MetaException, NoSuchObjectException {
+    setAlluxioLocation(partition);
+    return partition;
+  }
+
+  @Override
+  public List<String> filterPartitionNames(String dbName, String tblName,
+      List<String> partitionNames) throws MetaException {
+    return partitionNames;
+  }
+
+  @Override
+  public Index filterIndex(Index index) throws MetaException, NoSuchObjectException {
+    setAlluxioLocation(index.getSd());
+    return index;
+  }
+
+  @Override
+  public List<String> filterIndexNames(String dbName, String tblName, List<String> indexList) throws MetaException {
+    return indexList;
+  }
+
+  @Override
+  public List<Index> filterIndexes(List<Index> indexeList) throws MetaException {
+    for (Index index: indexeList) {
+      setAlluxioLocation(index.getSd());
+    }
+    return indexeList;
+  }
+
+  private void setAlluxioLocation(Table table) {
+    setAlluxioLocation(table.getSd());
+  }
+
+  private void setAlluxioLocation(Partition partition) {
+    setAlluxioLocation(partition.getSd());
+  }
+
+  private void setAlluxioLocation(StorageDescriptor sd) {
+    String location = sd.getLocation();
+    sd.setLocation(ALLUXIO_PREFIX + location);
```

Comment:
this is incorrect as it doesn't replace anything this will turn `s3://<path>` into `alluxio://s3://<path>`

Reply:
yes, it used to replace s3 with alluxio in the first version but I just kept it for testing and since in the tests we don't have s3 locations, I deleted the real logic.

Reply:
ok but this class is in src/main/java can we move it to src/test in that case. If in main it should be correct. If we move it just for test we should all reference to alluxio as that's unrelated then it's just a PrefixingMetastoreFilter in that case.

Reply:
yeah, I would have moved it to test but then it's not seen in the integration tests module. @massdosage suggested to create some test artifact, I'll try to do that

Reply:
maybe it's less work if we just make it generic and configurable and leave it in here as a reference implementation??

Reply:
might even be usuable

Reply:
Configurable how? In terms of location? I moved the filter to test and added the test artifact yesterday and it works. Also, maybe we can point to the Alluxio filter when/if that will be open-sourced as a reference implementation

Reply:
if it's move to test then that's fine as is. no problem!